aten/src/ATen/Parallel.h

```diff
@@ -7,6 +7,8 @@
 #include <exception>
 
 #ifdef _OPENMP
+#define INTRA_OP_PARALLEL
```

Comment:
Does it make sense to consider moving this into parallel_for, or having parallel_for switch to a serial implementation? This could also help remove a lot of TH code. I wonder if we currently have users who actively disable OpenMP and actually use these non-parallelized branches. Of course this is separate from simply setting the number of threads to 1.

In fact, maybe, being able to branch on the number of threads to a serial implementation within at::parallel could facilitate better code generation, because the compiler only needs to deal with a single for-loop instead and can avoid the OpenMP branch all-together.

Reply:
My take was that if user wants, for some reason, to remove parallel code from compilation completely, then I should support this, so I just simply replaced _OPENMP with INTRA_OP_PARALLEL. But I see your point, maybe we don't even need this too, I'll take another look at the use cases of _OPENMP, I think some of them could be removed and hidden inside parallel_for yes